addon/components/frost-password.js

```diff
@@ -4,36 +4,56 @@ import _ from 'lodash'
 
 export default Ember.Component.extend({
   layout: layout,
-  type: 'password',
   classNames: ['frost-password'],
-  revealable: false,
+  isCapsOn: false,
   showRevealer: false,
+  revealable: false,
   revealed: false,
-  timeoutInterval: 3000,
-  timeout: undefined,
-  prompt: Ember.computed('revealed', function () {
-    return this.get('revealed') ? 'hide' : 'show'
+  type: 'password',
+  wasCapsOn: false,
+
+  focusOut: Ember.on('focusOut', function () {
+    if (this.get('isCapsOn')) {
+      this.set('wasCapsOn', true)
+    }
+    this.set('isCapsOn', false)
+  }),
+  focusIn: Ember.on('focusIn', function () {
+    if (this.get('wasCapsOn')) {
+      this.set('isCapsOn', true)
+      this.set('wasCapsOn', false)
+    }
+  }),
+  keyDown: Ember.on('keyDown', function (e) {
+    var s = e || window.e
+    if (this.get('isCapsOn') === false && (s.which === 20 || s.keyCode === 20)) {
+      this.set('isCapsOn', true)
+    }
+  }),
+  keyUp: Ember.on('keyUp', function (e) {
+    var s = e || window.e
+    if (this.get('isCapsOn') === true && (s.which === 20 || s.keyCode === 20)) {
+      this.set('isCapsOn', false)
+    }
   }),
+  keyPressed: Ember.on('keyPress', function (e) {
+    var s = String.fromCharCode(e.which || e.keyCode)   // IE support
+    if ((s.toUpperCase() === s && s.toLowerCase() !== s && !e.shiftKey) ||
+      (s.toUpperCase() !== s && s.toLowerCase() === s && e.shiftKey)) { // caps is on
+      this.set('isCapsOn', true)
+    }
+  }),
+
   actions: {
-    toggleReveal () {
-      this.toggleProperty('revealed')
-      this.set('type', this.get('revealed') ? 'text' : 'password')
-      if (this.get('revealed')) {
-        this.startTimeout()
-      } else {
-        Ember.run.cancel(this.timeout)
-      }
-    },
     onInput (args) {
       this.set('showRevealer', args.value.length > 0)
```

Comment:
We should probably document proper variable/property naming based on team agreements. I vote :+1: on the `is` naming convention here though.